### ISSUE TITLE
rare crash on chunks view | line 2522, in draw_pieces_map | index out of range

### ISSUE THREAD

**Issue by gangelop** [tremc/tremc#32]
### The trace
```
Traceback (most recent call last):
  File "./tremc", line 3719, in <module>
    Interface()
  File "./tremc", line 987, in __init__
    self.run()
  File "./tremc", line 1090, in run
    self.draw_details()
  File "./tremc", line 2084, in draw_details
    self.draw_pieces_map(5)
  File "./tremc", line 2522, in draw_pieces_map
    block = (pieces[start >> 3]) << (start & 7)
IndexError: index out of range
```

### Steps to reproduce
1. add this torrent

        magnet:?xt=urn:btih:d640b4dcc12c08430456a2b900ae42cbbde8d79c&dn=chunks-reproducer

2. Navigate to the torrent and open the detailed view.
3. Press `t`,`c`. ==> crash

Same bug appears to affect `transmission-remote-cli` so this isn't something recent.

**Comment by gangelop** [comment]
I went and did some git archeology and here's what I've found so far.

It appears that this index out of range exception always existed.

* Originally, encountering the exception would cause the chunks view to not be drawn at all.
* Starting with fd2a63e, the exception is caught, the chunks view is drawn partially but instead of seeing the chunks you see the exception (as you might guess just by looking at the diff).

        Exception: <type 'exceptions.IndexError'>,string index out of range

* With `git log -G 'which details to display' --all` we can see that this area of code is touched one final time in `003ea42` where Benjamin decides to revert a bunch of different exception handlers, including the one we care about. However, at this point the IndexError exception causes the whole program to crash, which brings us to the current situation.

So what should we do now? This is the hard part.

We could just cherry pick `fd2a63e` and this would bring us back to drawing the exception in the chunks detail view instead of crashing.

But we should really figure out why the exception occurs. It should not happen at all.
I'll have to start by understanding how `draw_pieces_map()` works.
